src/benchmark.cc

```diff
@@ -781,7 +798,14 @@ void RunBenchmark(const benchmark::internal::Benchmark::Instance& b,
         report.cpu_accumulated_time = cpu_accumulated_time;
         report.bytes_per_second = bytes_per_second;
         report.items_per_second = items_per_second;
+        report.arg1 = b.arg1;
```

Comment:
ah, so you're assuming that arg1 and arg2 are used as loop indices, and only used in a single loop.

we have a couple of tests in basic_test that use the range loop twice (so it's already 2N) and examples in benchmark_test (BM_SetInsert) that use range_x to construct test data, then range_y to do the work. anything there based on assumptions of N and M will fail.

it might be worth making this more like the state methods SetItemsProcessed and SetBytesProcessed. SetComplexityN and SetComplexityM?